lib/util.js

```diff
@@ -280,14 +294,17 @@ function formatValue(ctx, value, recurseTimes) {
     }
   }
 
+  var constructor = getConstructorOf(value);
   var base = '', empty = false, braces, formatter;
 
   if (Array.isArray(value)) {
+    if (constructor === Array)
+      constructor = null;
```

Comment:
Null constructors aren't displayed. We explicitly null out for `Array` and `Object` so `{}` continues to display as `{}` (as opposed to `Object {}`) and `[]` continues to display as `[]` (as opposed to `Array []`). The reader should assume that if there is no constructor name that it is the default for the brace type. If we didn't do this the output would get pretty noisy for large object graphs, and we would have to modify a bunch of tests.